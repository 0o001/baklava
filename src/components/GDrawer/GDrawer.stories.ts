import { storiesOf } from '@storybook/vue';
import GDrawer from './GDrawer.vue';
import GButton from '../GButton';

storiesOf('GDrawer', module)
  .addParameters({ component: GDrawer })
  .add('Base', () => ({
    components: { GDrawer, GButton },
    data: () => ({
      show: false,
    }),
    methods: {
      onShow: function () {
        this.show = true;
      },
      onClose () {
        this.show = false;
      }
    },
    template: `
      <div class="g-p-10">
        <GDrawer
          url="https://m.wikipedia.org"
          title="Help"
          :show="show"
          @close="onClose"
          top=0
        >
        </GDrawer>
        <GButton @click="onShow" variant="primary">Open Drawer</GButton>
      </div>
    `,
  }))
