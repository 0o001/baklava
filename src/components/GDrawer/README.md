# GDrawer

## Props

| Name         | Type      | Description | Default     |
| ------------ | --------- | ----------- | ----------- |
| `url`        | `String`  |             | `""`        |
| `title`      | `String`  |             | `""`        |
| `show`       | `Boolean` |             | `false`     |
| `top`        | `Number`  |             | `100`       |
| `close`      | `Function`|             | `undefined` |