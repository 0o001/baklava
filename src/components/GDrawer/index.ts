import GDrawer from './GDrawer.vue';

export default GDrawer;
