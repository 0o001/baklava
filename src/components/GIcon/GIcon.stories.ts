import { storiesOf } from '@storybook/vue';
import GIcon from './GIcon.vue';

storiesOf('GIcon', module)
  .addParameters({ component: GIcon })
  .add('Example', () => ({
    components: { GIcon },
    template: `<div> <GIcon name="alert-circle" strokeWidth='2px' /> <GIcon stroke="black" size="44px" fill="var(--orange-100)" name="star" /> </div>`,
  }))