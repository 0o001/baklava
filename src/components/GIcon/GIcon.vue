<template>
  <svg
    v-bind="this.attrs"
    v-on="$listeners"
    :class="classNames"
    :dataName="this.name"
    :dataTags="tags"
    :dataType="this.name"
    v-html="icon.contents"
  />
</template>

<script lang="ts">

import feather from 'feather-icons';

export default {
  name: 'GIcon',
  props: {
    animation: {
      type: String,
      default: undefined,
    },
    animationSpeed: {
      type: String,
      default: undefined,
    },
    fill: {
      type: String,
      default: 'none',
    },
    size: {
      type: [Number, String],
      default: '24px',
    },
    stroke: {
      type: String,
      default: 'currentColor',
    },
    strokeLinecap: {
      type: String,
      default: 'round',
    },
    strokeLinejoin: {
      type: String,
      default: 'round',
    },
    strokeWidth: {
      type: [Number, String],
      default: 2,
    },
    tag: {
      type: String,
      default: 'i',
    },
    name: {
      type: String,
      required: true,
      validator (name) {
        if (!feather.icons[name]) {
          throw new Error(`"${name}" is not an available icon type.`);
        }
        return true;
      },
    },
    color: {
      type: String,
      default: undefined,
    },
  },
  computed: {
    icon () {
      return feather.icons[this.name];
    },
    tags () {
      return feather.icons[this.name] ? feather.icons[this.name].tags : '';
    },
    classNames () {
      const classes = {
        'g-icon': true,
        feather: true,
        [`feather--${this.name}`]: true,
        [`feather--${this.animation}`]: this.animation,
        [`feather--${this.animationSpeed}`]: this.animationSpeed,
      };
      return classes;
    },
    attrs () {
      return {
        ...this.icon.attrs,
        fill: this.fill,
        height: this.size,
        stroke: this.color ? `var(--${this.color})` : this.stroke,
        'stroke-linecap': this.strokeLinecap,
        'stroke-linejoin': this.strokeLinejoin,
        'stroke-width': this.strokeWidth,
        width: this.size,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
@keyframes feather--spin {
  from {
    transform: rotate(0);
  }
  to {
    transform: rotate(360deg);
  }
}
.feather {
  display: inline-block;
  overflow: hidden;
  &--spin {
    animation: feather--spin 2s linear infinite;
  }
  &--pulse {
    animation: feather--spin 2s infinite steps(8);
  }
  &--slow {
    animation-duration: 3s;
  }
  &--fast {
    animation-duration: 1s;
  }
  &__content {
    display: block;
    height: inherit;
    width: inherit;
  }
}
</style>
