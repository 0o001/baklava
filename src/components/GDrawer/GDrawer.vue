<template>
  <div
    v-bind="$attrs"
    v-on="$listeners"
    :class="classNames"
    :style="{ top: top + 'px', height: 'calc(100% - ' + top + 'px)' }"
    v-if="show"
  >
    <div
      class="g-drawer-content"
    >
      <div class="g-drawer-header">
        <div class="g-drawer-title">
          {{ title }}
        </div>
        <div
          @click="close"
          class="g-drawer-close-button"
        >
          <GIcon name="x" />
        </div>
      </div>
      <div class="g-drawer-body">
        <iframe
          :src="url"
          height="100%"
          width="100%"
          :title="title"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts">

import GIcon from '../GIcon';

export default {
  name: 'GDrawer',
  components: {
    GIcon,
  },
  props: {
    url: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: '',
    },
    show: {
      type: Boolean,
      default: false,
    },
    top: {
      type: Number,
      default: 100,
    },
  },
  methods: {
    close () {
      this.$emit('close');
    },
  },
  computed: {
    classNames () {
      return {
        'g-drawer': true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-drawer {
  position: fixed;
  right: 0;
  // top: 100px;
  // height: calc(100% - 100px);
  width: 424px;
  z-index: 100;

  .g-drawer-content {
    width: 100%;
    height: 100%;
    padding: 16px;
    box-shadow: 0px 8px 16px rgba(39, 49, 66, 0.2);

    .g-drawer-header {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-bottom: 8px;

      .g-drawer-title {
        font-size: 18px;
        line-height: 21px;
        font-weight: 500;
      }

      .g-drawer-close-button {
        cursor: pointer;
      }
    }

    .g-drawer-body {
      width: 100%;
      height: 100%;
      iframe {
        border-width: 0;
      }
    }
  }
}
</style>
