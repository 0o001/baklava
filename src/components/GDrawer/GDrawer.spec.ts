import { mount } from '@vue/test-utils';
import GDrawer from '.';
import GIcon from '../GIcon';

describe('GDrawer', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(GDrawer);
  });

  it('should match snapshot when show is prop is given as false ', () => {
    wrapper.setProps({
      url: '',
      title: 'Help',
      show: false,
      top: 0
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should props with default values when no props are set', () => {
    expect(wrapper.vm.classNames).toEqual({
      'g-drawer': true,
    });
  });

  it('should render prop set', async () => {
    // given

    wrapper.setProps({
      url: '',
      title: 'Help',
      show: true,
      top: 0
    });

    expect(wrapper.find('.g-drawer-content'))
    expect(wrapper.find('.g-drawer-title'))
  });
});
