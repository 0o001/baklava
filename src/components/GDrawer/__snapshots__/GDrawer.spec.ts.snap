// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDrawer should match snapshot when show is prop is given as false  1`] = `<!---->`;
